### ISSUE TITLE
Shellcheck does not allow for shebangs with a space

### ISSUE THREAD

**Issue by thibmaek** [gruntwork-io/pre-commit#47]
The custom regex check will fail if the shebang is in the format of `#! /usr/bin/bash` and only work when removing the space e.g `#!/usr/bin/bash`.

Spaces in shebangs are valid.

**Comment by brikis98** [comment]
Do we have a script with a space after the shebang somewhere?

**Comment by thibmaek** [comment]
Not in this repo maybe, but if I try to use this hook in a repo it will not work unless removing the space.
I tried to test by [changing the regex](https://github.com/gruntwork-io/pre-commit/blob/master/hooks/shellcheck.sh#L39) and validating on regexr but I'm not sure which format of regex that is (PCRE?)

**Comment by brikis98** [comment]
Ohhh, I think I misunderstood. What you're saying is that the regex we use to identify shell scripts is what would need to be fixed?

**Comment by thibmaek** [comment]
Yep indeed, the regex needs to option to allow a space between right after the bang symbol:

- regex validates: `#!/usr/bin/env bash`
- regex invalidates: `#! /usr/bin/env bash`

**Comment by brikis98** [comment]
Understood! And, indeed, [spaces after shebang are allowed](https://unix.stackexchange.com/questions/276751/is-space-allowed-between-and-bin-bash-in-shebang), as you mentioned. Would you be up for a PR to tweak the regex?

**Comment by thibmaek** [comment]
Yeah sure, do you know what format the regex is? I tried it on regexr but it didn't validate. Maybe its perl regex?

**Comment by brikis98** [comment]
It's grep format: https://stackoverflow.com/a/4192817/483528.
